config/initializers/warden_hooks.rb

```diff
@@ -0,0 +1,10 @@
+Warden::Manager.after_set_user do |user,auth,opts|
+  scope = opts[:scope]
+  auth.cookies.signed["#{scope}.id"] = user.id
+  auth.cookies.signed["#{scope}.expires_at"] = 30.minutes.from_now
+end
```

Comment:
after_set_user happens on each request like we were hoping

https://github.com/hassox/warden/wiki/Callbacks#after_set_user